React/nested-event-handling.md

```diff
@@ -180,187 +197,103 @@ The CSS is the most straightforward thing on our todo list, so let's get that ou
+Change the initial state in the `App` to verify this works as intended.
 
 #### The Event Handler
+
 Next, let's explore the event handler setup. What data do we need from the button click event to appropriately change the overall student's state?
 
-1. An Event Handler in the `Student.js` file
-1. A function that has access to the data (`this.state`) inside of `StudentCollection.js`
+1. An Event Handler in the `Student.js` file, we can use `onButtonClick` and `onFullNameInputChange`.
+1. A function that has access to the state inside of `App.js`.
 
-First we'll create the event handler function with an initial logging statement. We'll follow the same pattern as before for a DOM event handler: start with `on`, then indicate what element and what event are being handled.
+First we will modify the `onButtonClick` as follows.
 
 ```javascript
-// Student.js
-const Student = (props) => {
-  const onPresentButtonClick = () => {
-    console.log(props);
+const onButtonClick = () => {
+    const updatedStudent = {
+      fullName: props.fullName,
+      birthday: props.birthday,
+      email: props.email,
+      // Toggle present to a new value
+      present: !props.present,
+      id: props.id,
+    }
+
+    // call the function passed from `App`
+    props.onUpdateStudent(updatedStudent);
   }
-// ... the middle of Student.js
 ```
 
-Then we tie the button to the event handler function:
-
-```js
-// Student.js
-//...
-return (
-<button
-  disabled={ props.isPresent }
-  onClick={ onPresentButtonClick }
-  >Mark Present </button>
-  );
-}
-```
-
-Test it out and see what gets logged in the console.
-
 **Question:** What piece of data is going to help us determine exactly which student should be updated?
 
 <details>
 <summary>Answer</summary>
-The student we are interested in updating is something we can match using `props.index`. 
+The student we are interested in updating is something we can match using `props.id`. 
 </details>
 
-**Question:** Wait, why does this work at all? Shouldn't we lose access to `props` once the component is rendered?
+**Question:** Wait, why does this work at all? Shouldn't we lose access to `props` in the onButtonClick function due to scope?
 
 <details>
 <summary>Answer</summary>
-`onPresentButtonClick` is closed around `props`, so the `props` variable is still accessible! Remember, all we need to close is three things:
+`onButtonClick` is closed around `props`, so the `props` variable is still accessible! Remember, all we need to close is three things:
 
 1. Nest a function inside a function
 1. Reference a variable from the outer function in the inner function
 1. Make the inner function available outside the outer function
 
-1. `onPresentButtonClick` is nested inside of `Student`
-1. We reference `props` inside of `onPresentButtonClick`
-1. We hand `onPresentButtonClick` to the `<button>`, where it lives on after `Student` has finished running!
+1. `onButtonClick` is nested inside of `Student`
```

Comment:
I think this is intended to be 1,2,3 but it's rendering as 4,5,6. Perhaps an additional newline would fix this problem.

Reply:
Fixed, and thanks!